cni/pkg/nodeagent/net.go

```diff
@@ -50,15 +50,32 @@ var _ MeshDataplane = &NetServer{}
 // This cache will also be updated when the K8S informer gets a new pod.
 // This cache represents the "state of the world" of all enrolled pods on the node this agent
 // knows about, and will be sent to any connecting ztunnel as a startup message.
-func (s *NetServer) ConstructInitialSnapshot(ambientPods []*corev1.Pod) error {
+//
+// - For each of these existing snapshotted pods, steps into their netNS, and reconciles their
+// existing iptables rules against the expected set of rules. This is used to handle reconciling
+// iptables rule drift/changes between versions.
+func (s *NetServer) ConstructInitialSnapshot(existingAmbientPods []*corev1.Pod) error {
```

Comment:
Just a curiosity: Is there a guarantee that ConstructInitialSnapshot will always be executed before any other "unenroll events" are handled?

Reply:
Yes, explicitly, in all cases. If you look at `server.go` `Start()` you'll notice we call it before we start the informer.

Reply:
Thank you for the confirmation 👍